ansible_base/oauth2_provider/views/token.py

```diff
@@ -51,7 +51,41 @@ def create_token_response(self, request):
             return request.build_absolute_uri(), {}, str(e), e.status_code
 
 
+class OAuth2TokenPermission(permissions.BasePermission):
+    # An app token is a token that has an application attached to it
+    # A personal access token (PAT) is a token with no application attached to it
+    # With that in mind:
+    # - An app token can be read, changed, or deleted if:
+    #   - I am the superuser
+    #   - I am the admin of the organization of the application of the token
+    #   - I am the user of the token
+    # - An app token can be created if:
+    #   - I have read access to the application (currently this means: I am the superuser)
+    # - A PAT can be read, changed, or deleted if:
+    #   - I am the superuser
+    #   - I am the user of the token
+    # - A PAT can be created if:
+    #   - I am a user
+
+    def has_permission(self, request, view):
+        # Handle PAT and app token creation separately
+        if request.method == 'POST':
+            if request.data.get('application'):
+                # TODO: Change this once dab#424 is fixed
+                return request.user.is_superuser
+        return request.user.is_authenticated
+
+    def has_object_permission(self, request, view, obj):
+        if request.method in ['GET', 'PATCH', 'PUT', 'DELETE']:
+            if request.user.is_superuser:
+                return True
+            if obj.application and obj.application.organization.admins.filter(id=request.user.id).exists():
```

Comment:
I think we are working on nixing the `users/admins` fields off the Org/Team roles so this will likely need to change to a RoleDefinition or something like that.